src/main/java/frc/lib/team3015/subsystem/SubsystemFault.java

```diff
@@ -0,0 +1,68 @@
+package frc.lib.team3015.subsystem;
+
+import edu.wpi.first.wpilibj.Timer;
+import frc.lib.team6328.util.Alert;
+
+import java.util.Objects;
+
+public class SubsystemFault{
+  public final String description;
+  public final double timestamp;
+  public final boolean isWarning;
+  public final boolean sticky;
+  public SubsystemFault(String description, boolean isWarning) {
+    this(description, isWarning, false);
+  }
+
+  public SubsystemFault(String description) {
+    this(description, false);
+  }
+
+  // what does sticky mean specifically?
+  public SubsystemFault(String description, boolean isWarning, boolean sticky) {
+    this.description = description;
+    this.timestamp = Timer.getFPGATimestamp();
+    this.isWarning = isWarning;
+    this.sticky = sticky;
+
+    if(isWarning){
+      Alert alert = new Alert(description, Alert.AlertType.WARNING);
+
+      // is there a specific condition that we should look at for when to make this alert NOT show up?
```

Comment:
If a SubsystemFault object is created, the state of the alert should always be true. The only consideration if the severity of the alert (error vs. warning), which you have handled.

Reply:
Thinking more about this, this approach that I suggested would result in multiple Alerts being created for the same failure at a fixed interval. For example, if a TalonFX wasn't on the CAN bus, every second a new SystemFault, and, therefore, a new Alert, would be created. The FaultReporter class checks for duplicates, but Alert does not. Therefore, I think the creation of the Alert should be escalated to the FaultReporter class which is aware of duplicates. I'll leave a comment in that class as well.